lib/vector-buffers/src/variants/disk_v2/tests/invariants.rs

```diff
@@ -647,7 +647,16 @@ async fn writer_waits_for_reader_after_validate_last_write_fails_and_data_file_s
                 assert_buffer_size!(ledger, records_written, total_bytes_written);
             }
 
+            // Advance the time to ensure that we can trigger a full flush so that all writer bytes
+            // are demonstrably on disk after doing so.
+            tokio::time::pause();
+            tokio::time::advance(DEFAULT_FLUSH_INTERVAL).await;
+
+            writer.flush().await.expect("flush should not fail");
             writer.close();
+
+            tokio::time::resume();
+
```

Comment:
This test flaked for me locally, specifically around the expectation of the data file being a certain size before we go and mess with its contents. This is because while we flush after writes, we don't do a "full" flush (aka `fsync` to disk)` unless we've exceeded the flush interval (`500ms` by default) since our last full flush.

In tests, they can easily run faster than 500ms, or maybe the OS decides to flush to disk immediately... so this was inherently flaky. Now we're forcefully incrementing the time so we can do one final flush that will be triggered as a "full" flush, thus ensuring the expected data in the file is there and on disk.